Core/include/Acts/EventData/GenericBoundTrackParameters.hpp

```diff
@@ -156,11 +156,56 @@ class GenericBoundTrackParameters {
   ParametersVector& parameters() { return m_params; }
   /// Parameters vector.
   const ParametersVector& parameters() const { return m_params; }
+  /// Vector of impact parameters (i.e., d0 and z0 if nDim=2
+  /// or d0, z0, and t if nDim=3)
+  template <unsigned int nDim = 2>
+  const ActsVector<nDim>& impactParameters() const {
```

Comment:
This returns a reference to a local stack variable, so will dangle immediately.
You could return a map type, that points to the underlying `m_params` member variable, which would be safer.

Also: I'm wondering if separate names, like `impactParameters2()` and `impactParameters3()` would make clearer what's being accessed here.